### ISSUE TITLE
Bug with external Google contacts

### ISSUE THREAD

**Issue by Alecent** [jitsi/jitsi#323]
On Windows last version (including nightly build .11).

When we add an external google contact source (xxxxxx@gmail.com), we have the popup window to ask for autorization to use it, the URL is okay to the google security access, it gave the token code. But Jitsti seems to not save it and ask for the same procedure each time we try to search (or use, or display) a contact. So, first, it's boring, secondly, it is completely useless and unusable... :/

**Comment by ibauersachs** [comment]
@Alecent You can disable contact search for each XMPP account in the account's connection options.
@cobratbq Can you please have a look at this?

**Comment by Alecent** [comment]
@ibauersachs yes, thanks :) but I already did it. 

For additional information, I tried to delete all accounts (because I had two Google accounts), recreate one (with full process of token at google, ...) but it still doesn't function, whatever for one or two accounts. So, I'm waiting on you now :)

**Comment by Leyline77** [comment]
I found that: when I make the account name something simple (no @ sign like it shows) it does not ask me for the authentication token more than once.
However it still does not show me results from my google contacts when I search for them...

To clarify NOT account1@gmail.com
but instead account1_gmail

**Comment by cklein05** [comment]
Hi there,

I'm facing the same problem.

@ibauersachs You've assigned that issue to cobratbq. However, he wasn't active on GitHub since November 2016. Are you sure, that there will be any progress on that issue?

BTW, actually I'm trying to share my FRITZ!Box phone book. Since FRITZ!Box supports Google Contacts as well (and it has a working implementation!), the Google approach was just a workaround for getting FRITZ!Box contacts into Jitsi Desktop.

However, I consider writing a Jitsi contacts plugin for FRITZ!Box contacts. Are you interested in implementing this into the current upstream trunk?

In any case, I'm still interested in getting Google Contacts to work, since sharing with Google, for me,  provides many more advantages.

Carsten

**Comment by ibauersachs** [comment]
Afaik Danny is busy with other work, but he promised to take a look. As to when that is I can't say.

A Fritzbox-Plugin would be welcome. Please consider signing the CLA before you start working on it so that there aren't any surprises.

**Comment by cklein05** [comment]
Thanks for you reply. Signing the what? Not quite sure what CLA means (contributor license agreement?) Where do I find that document?

**Comment by damencho** [comment]
https://github.com/jitsi/jitsi#contributing points to: https://github.com/jitsi/jitsi/blob/master/CONTRIBUTING.md

**Comment by cobratbq** [comment]
Hi all, sorry for late reply. I have been quite busy ... partly with urgent things afk. (On previous comment, github doesn't seem to register activity on a fork as it assumes you'll merge it back and activity only seems to get registered after delivering back. https://github.com/cobratbq/otr4j/commits/master)

As for the Google Contacts thing. @damencho can you check the build process. We're supposed to provide an OAuth2 client id and secret for communications with Google servers and it seems that they are not provided. Start from the build script: google.api.client.id.

**Comment by damencho** [comment]
Yep, it is there -Dgoogle.api.client.id= and -Dgoogle.api.client.secret= in the windows build, I checked the scripts and the class file.

**Comment by cobratbq** [comment]
Okay, in that case I'll set up a local build and see up to where the Google Contacts functionality works. I'll follow up later.

**Comment by cobratbq** [comment]
By the way, is this also set up for the other builds?

**Comment by cklein05** [comment]
@cobratbq  didn't want to blame you or get you into a hurry... :-) I know, we're all busy men with regular jobs :)
@anyone
As you likely have read, I'planing to implement a FRITZ!Box Contacts plugin. Jitsi is still on Java 7, right? So, no Java 8 features should be used?

I'd like to use parts of an Apache 2.0 licensed library for TR-064 communication. That involves backporting to Java 7 as well as adding HTTPS support. Does anyone see any legal problems regarding this?

**Comment by cobratbq** [comment]
I'm not an expert, but Apache 2.0 licensed libraries should be fine.

**Comment by cobratbq** [comment]
Okay, it seems to be fixed with 2bde29d2f9e4e0f7b66b416232c759fb589058c6. I had to switch to a different json parsing library ... which kinda worries me as the original library is also used in the google client API that I depend on. So I can't fully guarantee that nothing weird may happen in the future in the same area of functionality.
For now, however, this should fix the problem. I've committed the changes to 'master', so hopefully the fix is available in the next nightly build.

Let me know (using a mention) if this works for you.

**Comment by cklein05** [comment]
Thanks for your work. I will check this tomorrow, Monday 10th.

**Comment by ibauersachs** [comment]
Apache 2.0 libraries are perfectly fine, as are the Eclipse Public License, LGPL, MIT, BSD, etc.

I'm not keen on starting to use Java 8 right now, but as our 2.12 likely won't be out before Debian Jessie is released, I also won't object switching to it anymore. You might want to raise that topic on the dev-mailing list again.

**Comment by cklein05** [comment]
@cobratbq now, the Google Contacts plugin works as expected :-) However, I can't say anything about the weird side effects you cannot guarantee to happen. So, we/you should keep our eyes on that.

**Comment by cobratbq** [comment]
@cklein05 Good to hear that everything's working again!

Don't worry. I don't expect you to figure out the weird side-effect. Basically, what it boils down to is when "suddenly" the OAuth2 support stops working, something might be wrong there. I'll need to look into whether I can update the Google Data API libraries. That might fix this and it might not be an issue at all. What's tricky is that it's hard to know for sure.

**Comment by jschwartzenberg** [comment]
> However, I consider writing a Jitsi contacts plugin for FRITZ!Box contacts. Are you interested in implementing this into the current upstream trunk?

Are you still considering this? I was thinking of the same thing. It would be nice to support the same features as the FRITZ!Fon application inside Jitsi.

**Comment by cklein05** [comment]
Julius, yes, I'm still considering this. However, I was quite busy the last weeks and likely will be some more. I'm on vacation for the next two weeks. Likely the plugin is getting kicked off in August...

My plans are to write a plugin similar to the Google Contacts plugin. However, this will likely lack some features available with the FRITZ!Fon app, since the Jitsi contacts plugin architecture only supports kind of contact lookup. Do you have other ideas to implement?

The preferred way of accessing the FB is the TR-64 protocoll (recommended by AVM). I've found a Java library on GitHub, which, however, is on Java 8 but can easily be ported back to Java 7. But this library is quite large an covers a lot of features that I don't need, so my plan is to heavily strip down that library, likely ending with a complete new (Jitsi) private implementation.

Additionally, my goal is to add some auto-discovery feature using the common fritz.box name, as well as some configurable filter capabilities in order to filter out certain contacts present in the FB (internal, external, system contacts etc.).

**Comment by jschwartzenberg** [comment]
> Do you have other ideas to implement?

When you use the FRITZ!Fon app, you can just enter the maintenance password of the FRTIZ!Box and it'll automatically create a SIP account and make an extra telephony device available. I think the same could be implemented with Jitsi as well, so that all you need to do to set it up:
1. Add account
2. Select FRITZ!Box in the network list
3. Enter the FRITZ!Box password

Another nice idea would be to be able to use the provisioning features that Jitsi already has so that Jitsi clients behind a FRITZ!Box could be provisioned fully automatically, but that would probably require working together with AVM/Freetz.

**Comment by code-chicken** [comment]
I am a bit confused: Is this issue still about Google Contacts or is it about Fritz!Box now?
If it is still about Google Contacts: Today I installed Jitsi and I am having exact the same problems as the original poster explained. With each letter that I enter there pops up a window and asks me for the auth code from google. It seems that the code is not being saved. This also happens if I run Jitsi as Administrator.
This is my environment:
Jitsi 2.10.5550
Java 1.8.0_151
Windows 10 (1709 with all updates applied)

If this is thread is not about Google Contacts any more I can create a new issue and delete this article.

![image](https://user-images.githubusercontent.com/3024019/35980896-282f0af2-0cec-11e8-9cc4-77caca0c2ea3.png)
